pkgs/applications/misc/sioyek/default.nix

```diff
@@ -25,29 +26,44 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-F71JXgYaWAye+nlSrZvGjJ4ucvHTx3tPZHRC5QI4QiU=";
   };
 
-  buildInputs = [ gumbo harfbuzz jbig2dec mupdf mujs openjpeg qt3d qtbase ];
+  buildInputs = [ gumbo harfbuzz jbig2dec mupdf mujs openjpeg qt3d qtbase ]
+    ++ lib.optionals stdenv.isDarwin [ freetype ];
 
   nativeBuildInputs = [ installShellFiles wrapQtAppsHook qmake ];
 
+  qmakeFlags = lib.optionals stdenv.isDarwin [ "CONFIG+=non_portable" ];
+
   postPatch = ''
+    substituteInPlace pdf_viewer_build_config.pro \
+      --replace "-lmupdf-threads" "-lgumbo -lharfbuzz -lfreetype -ljbig2dec -ljpeg -lopenjp2"
     substituteInPlace pdf_viewer/main.cpp \
       --replace "/usr/share/sioyek" "$out/share" \
       --replace "/etc/sioyek" "$out/etc"
   '';
 
-  postInstall = ''
+  postInstall = if stdenv.isLinux then ''
```

Comment:
This assumes that anything that's not `isLinux` is darwin and goes ahead and installs macos-specific stuff. I don't think we can make that assumption and it's much better to activate the macos-specific stuff when we explicitly detect darwin.

Reply:
Done

Reply:
This'll work, but I think what I meant was that darwin should have the special case, and all the other paths should follow the "linux" approach. Other platforms are likely to be much more like linux than macos.

Reply:
I see, updated